keys/keys.go

```diff
@@ -62,7 +59,7 @@ func NodeStatusKey(nodeID int32) proto.Key {
 
 // MakeNameMetadataKey returns the key for the namespace.
 func MakeNameMetadataKey(parentID uint32, name string) proto.Key {
-	k := make([]byte, 0, len(NameMetadataPrefix)+maxVarintSize+len(name))
+	k := make([]byte, 0, len(NameMetadataPrefix)+binary.MaxVarintLen64+len(name))
```

Comment:
The use of `binary.PutVarint` in `util/encoding.Encode` should really be removed. Looks like the only use of that method is by `engine.Increment()` which appears to be unused.